scripts/setup.sh

```
@@ -0,0 +1,86 @@
+    echo "❌ Docker is not installed."
+    echo "📥 Installing Docker..."
+    
+    # Detect OS and provide installation instructions
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        echo "   Please install Docker Desktop for Mac:"
+        echo "   https://docs.docker.com/desktop/mac/install/"
+    elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        echo "   Installing Docker on Linux..."
+        curl -fsSL https://get.docker.com -o get-docker.sh
+        sudo sh get-docker.sh
+        sudo usermod -aG docker $USER
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)